--- original
+++ modified
@@ -10,6 +10,8 @@
 
 from matplotlib import pyplot
 import kwant
+import latex
+import html
 
 # First, define the tight-binding system
 
@@ -73,8 +75,9 @@
 sys.attach_lead(lead1)
 
 # Plot it, to make sure it's OK
-
-kwant.plot(sys)
+size = (latex.figwidth_in, 0.3 * latex.figwidth_in)
+kwant.plot(sys, file="1-quantum_wire_sys.pdf", fig_size=size, dpi=html.dpi)
+kwant.plot(sys, file="1-quantum_wire_sys.png", fig_size=size, dpi=html.dpi)
 
 # Finalize the system
 
@@ -98,8 +101,13 @@
 # Use matplotlib to write output
 # We should see conductance steps
 
-pyplot.figure()
+fig = pyplot.figure()
 pyplot.plot(energies, data)
-pyplot.xlabel("energy [in units of t]")
-pyplot.ylabel("conductance [in units of e^2/h]")
-pyplot.show()
+pyplot.xlabel("energy [in units of t]", fontsize=latex.mpl_label_size)
+pyplot.ylabel("conductance [in units of e^2/h]", fontsize=latex.mpl_label_size)
+pyplot.setp(fig.get_axes()[0].get_xticklabels(), fontsize=latex.mpl_tick_size)
+pyplot.setp(fig.get_axes()[0].get_yticklabels(), fontsize=latex.mpl_tick_size)
+fig.set_size_inches(latex.mpl_width_in, latex.mpl_width_in * 3. / 4.)
+fig.subplots_adjust(left=0.15, right=0.95, top=0.95, bottom=0.15)
+fig.savefig("1-quantum_wire_result.pdf")
+fig.savefig("1-quantum_wire_result.png", dpi=html.dpi)
