--- original
+++ modified
@@ -17,6 +17,7 @@
 
 # For plotting
 from matplotlib import pyplot
+import latex, html
 
 tau_x = tinyarray.array([[0, 1], [1, 0]])
 tau_z = tinyarray.array([[1, 0], [0, -1]])
@@ -46,12 +47,19 @@
     energies = kwant.physics.Bands(lead)
     energy_list = [energies(k) for k in momenta]
 
-    pyplot.figure()
+    fig = pyplot.figure()
     pyplot.plot(momenta, energy_list)
     pyplot.xlabel("momentum [in untis of (lattice constant)^-1]")
     pyplot.ylabel("energy [in units of t]")
     pyplot.ylim([-0.8, 0.8])
-    pyplot.show()
+    pyplot.setp(fig.get_axes()[0].get_xticklabels(),
+               fontsize=latex.mpl_tick_size)
+    pyplot.setp(fig.get_axes()[0].get_yticklabels(),
+               fontsize=latex.mpl_tick_size)
+    fig.set_size_inches(latex.mpl_width_in, latex.mpl_width_in * 3. / 4.)
+    fig.subplots_adjust(left=0.15, right=0.95, top=0.95, bottom=0.15)
+    fig.savefig("5-superconductor_band_structure_result.pdf")
+    fig.savefig("5-superconductor_band_structure_result.png", dpi=html.dpi)
 
 
 def main():
