--- original
+++ modified
@@ -10,6 +10,7 @@
 
 # For plotting.
 from matplotlib import pyplot
+import _defs
 
 def make_lead(a=1, t=1.0, W=10):
     # Start with an empty lead with a single square lattice
@@ -33,10 +34,19 @@
 
 def main():
     lead = make_lead().finalized()
-    kwant.plotter.bands(lead, show=False)
-    pyplot.xlabel("momentum [(lattice constant)^-1]")
-    pyplot.ylabel("energy [t]")
-    pyplot.show()
+    fig = kwant.plotter.bands(lead, show=False)
+    pyplot.xlabel("momentum [(lattice constant)^-1]",
+                 fontsize=_defs.mpl_label_size)
+    pyplot.ylabel("energy [t]", fontsize=_defs.mpl_label_size)
+    pyplot.setp(fig.get_axes()[0].get_xticklabels(),
+               fontsize=_defs.mpl_tick_size)
+    pyplot.setp(fig.get_axes()[0].get_yticklabels(),
+               fontsize=_defs.mpl_tick_size)
+    fig.set_size_inches(_defs.mpl_width_in, _defs.mpl_width_in * 3. / 4.)
+    fig.subplots_adjust(left=0.15, right=0.95, top=0.95, bottom=0.15)
+    for extension in ('pdf', 'png'):
+        fig.savefig("band_structure_result." + extension, dpi=_defs.dpi)
+
 
 
 # Call the main function if the script gets executed (as opposed to imported).
