--- original
+++ modified
@@ -18,6 +18,7 @@
 
 # For plotting
 from matplotlib import pyplot
+import _defs
 
 
 def make_system(a=1, t=1.0, r=10):
@@ -70,32 +71,43 @@
 
         energies.append(ev)
 
-    pyplot.figure()
+    fig = pyplot.figure()
     pyplot.plot(Bfields, energies)
-    pyplot.xlabel("magnetic field [some arbitrary units]")
-    pyplot.ylabel("energy [in units of t]")
-    pyplot.show()
+    pyplot.xlabel("magnetic field [some arbitrary units]",
+                 fontsize=_defs.mpl_label_size)
+    pyplot.ylabel("energy [in units of t]", fontsize=_defs.mpl_label_size)
+    pyplot.setp(fig.get_axes()[0].get_xticklabels(),
+               fontsize=_defs.mpl_tick_size)
+    pyplot.setp(fig.get_axes()[0].get_yticklabels(),
+               fontsize=_defs.mpl_tick_size)
+    fig.set_size_inches(_defs.mpl_width_in, _defs.mpl_width_in * 3. / 4.)
+    fig.subplots_adjust(left=0.15, right=0.95, top=0.95, bottom=0.15)
+    fig.savefig("closed_system_result.pdf")
+    fig.savefig("closed_system_result.png", dpi=_defs.dpi)
 
 
 def plot_wave_function(sys):
     # We reset the magnetic field to equal to 0.
     global B
     B = 0.
+    size = (_defs.figwidth_in, _defs.figwidth_in)
 
     # Calculate the wave functions in the system.
     ham_mat = sys.hamiltonian_submatrix(sparse=True)
     evecs = sla.eigsh(ham_mat, k=20, which='SM')[1]
 
     # Plot the probability density of the 10th eigenmode.
-    kwant.plotter.map(sys, np.abs(evecs[:, 9])**2, colorbar=False)
+    kwant.plotter.map(sys, np.abs(evecs[:, 9])**2, colorbar=False,
+                      file="closed_system_eigenvector.pdf",
+                      fig_size=size, dpi=_defs.dpi)
+    kwant.plotter.map(sys, np.abs(evecs[:, 9])**2, colorbar=False,
+                      file="closed_system_eigenvector.png",
+                      fig_size=size, dpi=_defs.dpi)
 
 
 def main():
     sys = make_system()
 
-    # Check that the system looks as intended.
-    kwant.plot(sys)
-
     # Finalize the system.
     sys = sys.finalized()
 
@@ -108,6 +120,7 @@
     sys = make_system(r=30).finalized()
     plot_wave_function(sys)
 
+
 # Call the main function if the script gets executed (as opposed to imported).
 # See <http://docs.python.org/library/__main__.html>.
 if __name__ == '__main__':
