--- original
+++ modified
@@ -1,4 +1,4 @@
-
+import _defs
 from cmath import exp
 import numpy as np
 import kwant
@@ -11,6 +11,13 @@
 
 ######## What is a Site? ##############
 
+def save_figure(file_name):
+    if not file_name:
+        return
+    for extension in ('pdf', 'png'):
+        plt.savefig('.'.join((file_name,extension)),
+                    dpi=_defs.dpi, bbox_inches='tight')
+
 a = 1
 lat = kwant.lattice.square(a)
 syst = kwant.Builder()
@@ -19,6 +26,7 @@
 syst[lat(1,1)] = 4
 
 kwant.plot(syst)
+save_figure('FAQ122')
 
 
 
@@ -40,7 +48,14 @@
 syst[lat2(0,0)] = 4 ## syst[subB(0,0)] = 4
 
 kwant.plot(syst)
+save_figure('FAQ123')
 
+def save_figure(file_name):
+    if not file_name:
+        return
+    for extension in ('pdf', 'png'):
+        plt.savefig('.'.join((file_name,extension)),
+                    dpi=_defs.dpi, bbox_inches='tight')
 
 
 ########## What is a hopping? #######################
@@ -54,6 +69,7 @@
 syst[(lat(1,0) , lat(1,1))] = -1
 
 kwant.plot(syst)
+save_figure('FAQ124')
 
 
 
@@ -73,7 +89,7 @@
 
 syst[ (lat(i,j,k) for i in range(L) for j in range(W) for k in range(H)) ] = 4
 kwant.plot(syst)
-
+save_figure('FAQ1')
 
 ## Deleting sites to create a hole
 
@@ -84,6 +100,7 @@
             if ((L-2)*a/4 <= x <= 3*L*a/4) and (W*a/4 <= y <= 3*(W-2)*a/4) and (0 <= z <= (H-1)*a):
                 del syst[lat(i,j,k)]
 kwant.plot(syst)
+save_figure('FAQ2')
 
 
 
@@ -145,7 +162,7 @@
 
 ## Plotting the system
 plot_system(syst)
-
+save_figure('FAQ6B')
 
 
 
@@ -162,7 +179,7 @@
 
 syst[kwant.builder.HoppingKind((1,0), lat)] = -1
 kwant.plot(syst)
-
+save_figure('FAQ3')
 ## Polyatomic lattice
 
 lat = kwant.lattice.kagome()
@@ -193,12 +210,13 @@
 
 syst[kwant.builder.HoppingKind((0,1), b, b)] = -1 # equivalent to syst[kwant.builder.HoppingKind((0,1), b)] = -1
 plot_system(syst)
+save_figure('FAQ10')
 del syst[kwant.builder.HoppingKind((0,1), b, b)] ## delete the hoppings previously created
 syst[kwant.builder.HoppingKind((0,0), a, b)] = -1
 syst[kwant.builder.HoppingKind((0,0), a, c)] = -1
 syst[kwant.builder.HoppingKind((0,0), c, b)] = -1
 plot_system(syst)
-
+save_figure('FAQ11')
 
 
 ########## How to create the hoppings between adjacent sites? ################
@@ -213,11 +231,13 @@
 
 syst[lat.neighbors()] = -1 ## equivalent to lat.neighbors(1)
 kwant.plot(syst)
+save_figure('FAQ4')
 
 del syst[lat.neighbors()] ## deletes every hoppings previously created to add new one
 syst[lat.neighbors(2)] = -1
 
 kwant.plot(syst)
+save_figure('FAQ5')
 
 ## Polyatomic lattice
 
@@ -236,14 +256,17 @@
 ## .neighbors()
 syst[lat.neighbors()] = -1
 plot_system(syst)
+save_figure('FAQ7')
 del syst[lat.neighbors()] ## delete the hoppings previously created
 syst[a.neighbors()] = -1
 plot_system(syst)
+save_figure('FAQ8')
 del syst[a.neighbors()] ## deletes every hoppings previously created to add new one
 
 
 syst[lat.neighbors(2)] = -1
 plot_system(syst)
+save_figure('FAQ9')
 del syst[lat.neighbors(2)]
 
 
@@ -282,9 +305,10 @@
 syst[ (subB(i,j) for i in range(L) for j in range(W)) ] = 4
 syst[lat.neighbors()] = -1
 plot_system(syst)
+save_figure('FAQAB')
 ## We manually add sites of the same lead lattice
 
-
+a = 2
 lat2 = kwant.lattice.square(a)
 
 def shapetop(pos):
@@ -303,6 +327,7 @@
 syst[((lat2(i,-1), subA(i,0)) for i in range(5))] = -1
 syst[((lat2(i+2,5), subB(i,4)) for i in range(5))] = -1
 plot_system(syst)
+save_figure('FAQAC')
 ## Creation of the top lead
 
 lat_lead = kwant.lattice.square(a)
@@ -320,6 +345,8 @@
 
 syst.attach_lead(lead1)
 plot_system(syst)
+save_figure('FAQAD')
+
 ## Creation of the bottom lead
 
 sym_lead2 = kwant.TranslationalSymmetry((0,-a))
@@ -335,6 +362,7 @@
 syst.attach_lead(lead2)
 
 plot_system(syst)
+save_figure('FAQAE')
 
 
 
@@ -356,6 +384,7 @@
 cuboid.fill(model, cuboid_shape, (0, 0, 0));
 
 kwant.plot(cuboid);
+save_figure('FAQaaa')
 
 # Build electrode (black).
 def electrode_shape(site):
@@ -369,7 +398,7 @@
 
 cuboid.attach_lead(electrode)
 kwant.plot(cuboid);
-
+save_figure('FAQbbb')
 
 ###### How to extract the wavefunction informations on a specific site? ###############
 
@@ -389,6 +418,7 @@
 syst.attach_lead(lead)
 syst.attach_lead(lead.reversed())
 kwant.plot(syst)
+save_figure('FAQBA')
 
 fsyst = syst.finalized()
 
@@ -401,6 +431,7 @@
 plt.xlabel("momentum [(lattice constant)^-1]")
 plt.ylabel("energy [t]")
 plt.show()
+save_figure('FAQBB')
 plt.clf()
 
 wf = kwant.wave_function(fsyst, Ef)
@@ -474,7 +505,7 @@
 
     # Check that the system looks as intended.
 kwant.plot(syst)
-
+save_figure('FAQTT')
     # Finalize the system.
 syst = syst.finalized()
 
@@ -488,7 +519,7 @@
 plt.xlabel("momentum [(lattice constant)^-1]")
 plt.ylabel("energy [t]")
 plt.show()
-
+save_figure('FAQSS')
 plt.clf()
 
 wf = kwant.wave_function(syst, Ef)
