--- original
+++ modified
@@ -1,6 +1,7 @@
 # Frequently Asked Questions
 # ==========================
 
+import _defs
 import kwant
 import numpy as np
 import tinyarray
@@ -9,6 +10,12 @@
 matplotlib.rcParams['figure.figsize'] = (3.5, 3.5)
 
 
+def save_figure(file_name):
+    for extension in ('pdf', 'png'):
+        plt.savefig('.'.join((file_name, extension)),
+                    dpi=_defs.dpi, bbox_inches='tight')
+
+
 ######## What is a Site? ##############
 
 a = 1
@@ -19,6 +26,7 @@
 syst[lat(1, 1)] = 4
 
 kwant.plot(syst)
+save_figure("FAQ122")
 
 
 
@@ -40,6 +48,7 @@
 syst[lat2(0, 0)] = 4  # syst[subB(0, 0)] = 4
 
 kwant.plot(syst)
+save_figure("FAQ123")
 
 
 ########## What is a hopping? #######################
@@ -53,6 +62,7 @@
 syst[(lat(1, 0), lat(1, 1))] = -1
 
 kwant.plot(syst)
+save_figure("FAQ124")
 
 ########### How to make a hole in a system? ###################"
 
@@ -69,6 +79,7 @@
 
 syst[(lat(i, j, k) for i in range(L) for j in range(W) for k in range(H))] = 4
 kwant.plot(syst)
+save_figure("FAQ1")
 
 # Delete sites to create a hole
 
@@ -80,6 +91,7 @@
     del syst[site]
 
 kwant.plot(syst)
+save_figure("FAQ2")
 
 
 ################ How can we get access to the sites of our system? ####################
@@ -133,7 +145,7 @@
 
 ## Plotting the system
 plot_system(syst)
-
+save_figure("FAQ6B")
 
 
 ############### How to create every hoppings in a given direction using Hoppingkind? ################
@@ -149,6 +161,7 @@
 
 syst[kwant.builder.HoppingKind((1, 0), lat)] = -1
 kwant.plot(syst)
+save_figure("FAQ3")
 
 # Polyatomic lattice
 
@@ -181,12 +194,14 @@
 # equivalent to syst[kwant.builder.HoppingKind((0, 1), b)] = -1
 syst[kwant.builder.HoppingKind((0, 1), b, b)] = -1
 plot_system(syst)
+save_figure("FAQ10")
 # Delete the hoppings previously created
 del syst[kwant.builder.HoppingKind((0, 1), b, b)]
 syst[kwant.builder.HoppingKind((0, 0), a, b)] = -1
 syst[kwant.builder.HoppingKind((0, 0), a, c)] = -1
 syst[kwant.builder.HoppingKind((0, 0), c, b)] = -1
 plot_system(syst)
+save_figure("FAQ11")
 
 
 ########## How to create the hoppings between adjacent sites? ################
@@ -201,11 +216,13 @@
 
 syst[lat.neighbors()] = -1  # Equivalent to lat.neighbors(1)
 kwant.plot(syst)
+save_figure("FAQ4")
 
 del syst[lat.neighbors()]  # Delete all nearest-neighbor hoppings
 syst[lat.neighbors(2)] = -1
 
 kwant.plot(syst)
+save_figure("FAQ5")
 
 # Polyatomic lattice
 
@@ -224,14 +241,17 @@
 
 syst[lat.neighbors()] = -1
 plot_system(syst)
+save_figure("FAQ7")
 del syst[lat.neighbors()]  # Delete the hoppings previously created
 syst[a.neighbors()] = -1
 plot_system(syst)
+save_figure("FAQ8")
 del syst[a.neighbors()]  # Delete the hoppings previously created
 
 
 syst[lat.neighbors(2)] = -1
 plot_system(syst)
+save_figure("FAQ9")
 del syst[lat.neighbors(2)]
 
 
@@ -264,6 +284,7 @@
 syst[(subB(i, j) for i in range(L) for j in range(W))] = 4
 syst[lat.neighbors()] = -1
 plot_system(syst)
+save_figure("FAQAA")
 
 # Create a lead
 lat_lead = kwant.lattice.square()
@@ -273,6 +294,7 @@
 lead1[(lat_lead(i, 0) for i in range(2, 7))] = 4
 lead1[lat_lead.neighbors()] = -1
 plot_system(lead1)
+save_figure("FAQAB")
 
 syst[(lat_lead(i, 5) for i in range(2, 7))] = 4
 syst[lat_lead.neighbors()] = -1
@@ -280,9 +302,11 @@
 # Manually attach sites from graphene to square lattice
 syst[((lat_lead(i+2, 5), subB(i, 4)) for i in range(5))] = -1
 plot_system(syst)
+save_figure("FAQAC")
 
 syst.attach_lead(lead1)
 plot_system(syst)
+save_figure("FAQAD")
 
 
 ############# How to cut a finite system out of a system with translationnal symmetries? ###########
@@ -302,6 +326,7 @@
 cuboid = kwant.Builder()
 cuboid.fill(model, cuboid_shape, (0, 0, 0));
 kwant.plot(cuboid);
+save_figure("FAQaaa")
 
 # Build electrode (black).
 def electrode_shape(site):
@@ -316,6 +341,7 @@
 
 cuboid.attach_lead(electrode)
 kwant.plot(cuboid);
+save_figure("FAQbbb")
 
 
 ###### How does Kwant order the propagating modes of a lead? ######
@@ -348,6 +374,7 @@
     plt.show()
 
 plot_and_label_modes(flead, E)
+save_figure('FAQPM')
 plt.clf()
 
 # More involved example
@@ -363,6 +390,7 @@
 flead2 = lead2.finalized()
 
 plot_and_label_modes(flead2, 1)
+save_figure('FAQPMC')
 plt.clf()
 
 
@@ -395,7 +423,8 @@
 
 idx = syst.id_by_site[lat(0, 0)]  # look up index of site
 
-print('wavefunction on lat(0, 0): ', wf[idx])
+with open('FAQORD1.txt', 'w') as f:
+    print('wavefunction on lat(0, 0): ', wf[idx], file=f)
 
 lat = kwant.lattice.square(norbs=2)
 syst = make_system(lat)
@@ -408,4 +437,5 @@
 # to degrees of freedom on the same site.
 wf = wf.reshape(-1, 2)
 
-print('wavefunction on lat(0, 0): ', wf[idx])
+with open('FAQORD2.txt', 'w') as f:
+    print('wavefunction on lat(0, 0): ', wf[idx], file=f)
