--- original
+++ modified
@@ -1,11 +1,19 @@
 import math
 import cmath
 
+import _defs
 import matplotlib.pyplot as plt
 
 import kwant
 
 
+figsize = (_defs.figwidth_in, _defs.figwidth_in)
+def save_plot(fname):
+    for extension in ('.pdf', '.png'):
+        plt.savefig(fname + extension,
+                    size=figsize, dpi=_defs.dpi, bbox_inches='tight')
+
+
 def hopping(sitei, sitej, phi):
     xi, yi = sitei.pos
     xj, yj = sitej.pos
@@ -42,7 +50,9 @@
 def main():
     syst = make_system()
 
-    kwant.plot(syst)
+    fig, ax = plt.subplots(1, 1)
+    kwant.plot(syst, ax=ax)
+    save_plot('plot_qpc_syst')
 
     params = dict(phi=1/40, salt="")
     psi = kwant.wave_function(syst, energy=0.15, params=params)(0)
@@ -52,9 +62,13 @@
     density = sum(D(p) for p in psi)
     current = sum(J(p) for p in psi)
 
-    kwant.plotter.map(syst, density, method='linear')
-
-    kwant.plotter.current(syst, current)
+    fig, ax = plt.subplots(1, 1, figsize=figsize)
+    kwant.plotter.map(syst, density, method='linear', ax=ax)
+    save_plot('plot_qpc_density')
+
+    fig, ax = plt.subplots(1, 1, figsize=figsize)
+    kwant.plotter.current(syst, current, ax=ax)
+    save_plot('plot_qpc_current')
 
 
 if __name__ == '__main__':
