--- original
+++ modified
@@ -1,3 +1,4 @@
+import _defs
 import kwant
 from matplotlib import pyplot
 
@@ -22,7 +23,10 @@
     # checking shapes:
     sys = make_cuboid()
 
-    kwant.plot(sys)
+    size = (_defs.figwidth_in, _defs.figwidth_in)
+    for extension in ('pdf', 'png'):
+        kwant.plot(sys, file="plot_zincblende_sys1." + extension,
+                   fig_size=size, dpi=_defs.dpi)
 
     # visualize the crystal structure better for a very small system
     sys = make_cuboid(a=1.5, b=1.5, c=1.5)
@@ -30,8 +34,12 @@
     def family_colors(site):
         return 'r' if site.family == a else 'g'
 
-    kwant.plot(sys, site_size=0.18, site_lw=0.01, hop_lw=0.05,
-               site_color=family_colors)
+    size = (_defs.figwidth_in, _defs.figwidth_in)
+    for extension in ('pdf', 'png'):
+        kwant.plot(sys, site_size=0.18, site_lw=0.01, hop_lw=0.05,
+                   site_color=family_colors,
+                   file="plot_zincblende_sys2." + extension,
+                   fig_size=size, dpi=_defs.dpi)
 
 
 # Call the main function if the script gets executed (as opposed to imported).
