--- original
+++ modified
@@ -9,6 +9,7 @@
 # --------------------------
 #  - demonstrate different ways of plotting in 3D
 
+import _defs
 import kwant
 from matplotlib import pyplot
 
@@ -33,7 +34,10 @@
     # checking shapes:
     syst = make_cuboid()
 
-    kwant.plot(syst)
+    size = (_defs.figwidth_in, _defs.figwidth_in)
+    for extension in ('pdf', 'png'):
+        kwant.plot(syst, file="plot_zincblende_syst1." + extension,
+                   fig_size=size, dpi=_defs.dpi)
 
     # visualize the crystal structure better for a very small system
     syst = make_cuboid(a=1.5, b=1.5, c=1.5)
@@ -41,8 +45,12 @@
     def family_colors(site):
         return 'r' if site.family == a else 'g'
 
-    kwant.plot(syst, site_size=0.18, site_lw=0.01, hop_lw=0.05,
-               site_color=family_colors)
+    size = (_defs.figwidth_in, _defs.figwidth_in)
+    for extension in ('pdf', 'png'):
+        kwant.plot(syst, site_size=0.18, site_lw=0.01, hop_lw=0.05,
+                   site_color=family_colors,
+                   file="plot_zincblende_syst2." + extension,
+                   fig_size=size, dpi=_defs.dpi)
 
 
 # Call the main function if the script gets executed (as opposed to imported).
