--- original
+++ modified
@@ -6,6 +6,7 @@
 # --------------------------
 #  - Functions as values in Builder
 
+import _defs
 import kwant
 
 # For plotting
@@ -52,19 +53,25 @@
         smatrix = kwant.smatrix(sys, energy, args=[-welldepth])
         data.append(smatrix.transmission(1, 0))
 
-    pyplot.figure()
+    fig = pyplot.figure()
     pyplot.plot(welldepths, data)
-    pyplot.xlabel("well depth [t]")
-    pyplot.ylabel("conductance [e^2/h]")
-    pyplot.show()
+    pyplot.xlabel("well depth [t]",
+                 fontsize=_defs.mpl_label_size)
+    pyplot.ylabel("conductance [e^2/h]",
+                 fontsize=_defs.mpl_label_size)
+    pyplot.setp(fig.get_axes()[0].get_xticklabels(),
+               fontsize=_defs.mpl_tick_size)
+    pyplot.setp(fig.get_axes()[0].get_yticklabels(),
+               fontsize=_defs.mpl_tick_size)
+    fig.set_size_inches(_defs.mpl_width_in, _defs.mpl_width_in * 3. / 4.)
+    fig.subplots_adjust(left=0.15, right=0.95, top=0.95, bottom=0.15)
+    for extension in ('pdf', 'png'):
+        fig.savefig("quantum_well_result." + extension, dpi=_defs.dpi)
 
 
 def main():
     sys = make_system()
 
-    # Check that the system looks as intended.
-    kwant.plot(sys)
-
     # Finalize the system.
     sys = sys.finalized()
 
