--- original
+++ modified
@@ -11,6 +11,7 @@
 #  - Making scattering region and leads
 #  - Using the simple sparse solver for computing Landauer conductance
 
+import _defs
 from matplotlib import pyplot
 import kwant
 
@@ -69,7 +70,10 @@
 sys.attach_lead(right_lead)
 
 # Plot it, to make sure it's OK
-kwant.plot(sys)
+size = (_defs.figwidth_in, 0.3 * _defs.figwidth_in)
+for extension in ('pdf', 'png'):
+    kwant.plot(sys, file="quantum_wire_sys." + extension,
+               fig_size=size, dpi=_defs.dpi)
 
 # Finalize the system
 sys = sys.finalized()
@@ -90,8 +94,13 @@
 
 # Use matplotlib to write output
 # We should see conductance steps
-pyplot.figure()
+fig = pyplot.figure()
 pyplot.plot(energies, data)
-pyplot.xlabel("energy [t]")
-pyplot.ylabel("conductance [e^2/h]")
-pyplot.show()
+pyplot.xlabel("energy [t]", fontsize=_defs.mpl_label_size)
+pyplot.ylabel("conductance [e^2/h]", fontsize=_defs.mpl_label_size)
+pyplot.setp(fig.get_axes()[0].get_xticklabels(), fontsize=_defs.mpl_tick_size)
+pyplot.setp(fig.get_axes()[0].get_yticklabels(), fontsize=_defs.mpl_tick_size)
+fig.set_size_inches(_defs.mpl_width_in, _defs.mpl_width_in * 3. / 4.)
+fig.subplots_adjust(left=0.15, right=0.95, top=0.95, bottom=0.15)
+for extension in ('pdf', 'png'):
+    fig.savefig("quantum_wire_result." + extension, dpi=_defs.dpi)
