--- original
+++ modified
@@ -17,6 +17,7 @@
 
 # For plotting
 from matplotlib import pyplot
+import _defs
 
 tau_x = tinyarray.array([[0, 1], [1, 0]])
 tau_z = tinyarray.array([[1, 0], [0, -1]])
@@ -46,11 +47,19 @@
     # Make system and finalize it right away.
     lead = make_lead().finalized()
 
-    kwant.plotter.bands(lead, momenta=np.linspace(-1.5, 1.5, 101), show=False)
+    fig = kwant.plotter.bands(lead, momenta=np.linspace(-1.5, 1.5, 101),
+                              show=False)
     pyplot.xlabel("momentum [in untis of (lattice constant)^-1]")
     pyplot.ylabel("energy [in units of t]")
     pyplot.ylim([-0.8, 0.8])
-    pyplot.show()
+    pyplot.setp(fig.get_axes()[0].get_xticklabels(),
+               fontsize=_defs.mpl_tick_size)
+    pyplot.setp(fig.get_axes()[0].get_yticklabels(),
+               fontsize=_defs.mpl_tick_size)
+    fig.set_size_inches(_defs.mpl_width_in, _defs.mpl_width_in * 3. / 4.)
+    fig.subplots_adjust(left=0.15, right=0.95, top=0.95, bottom=0.15)
+    fig.savefig("superconductor_band_structure_result.pdf")
+    fig.savefig("superconductor_band_structure_result.png", dpi=_defs.dpi)
 
 
 # Call the main function if the script gets executed (as opposed to imported).
