--- original
+++ modified
@@ -13,6 +13,7 @@
 #  - Main motivation is to contrast to the implementation of superconductivity
 #    in tutorial5b.py
 
+import _defs
 import kwant
 
 import numpy as np
@@ -49,11 +50,20 @@
     # Make system and finalize it right away.
     lead = make_lead().finalized()
 
-    kwant.plotter.bands(lead, momenta=np.linspace(-1.5, 1.5, 101), show=False)
+    fig = kwant.plotter.bands(lead, momenta=np.linspace(-1.5, 1.5, 101),
+                              show=False)
     pyplot.xlabel("momentum [(lattice constant)^-1]")
     pyplot.ylabel("energy [t]")
     pyplot.ylim([-0.8, 0.8])
-    pyplot.show()
+    pyplot.setp(fig.get_axes()[0].get_xticklabels(),
+               fontsize=_defs.mpl_tick_size)
+    pyplot.setp(fig.get_axes()[0].get_yticklabels(),
+               fontsize=_defs.mpl_tick_size)
+    fig.set_size_inches(_defs.mpl_width_in, _defs.mpl_width_in * 3. / 4.)
+    fig.subplots_adjust(left=0.15, right=0.95, top=0.95, bottom=0.15)
+    for extension in ('pdf', 'png'):
+        fig.savefig("superconductor_band_structure_result." + extension,
+                    dpi=_defs.dpi)
 
 
 # Call the main function if the script gets executed (as opposed to imported).
