--- original
+++ modified
@@ -10,6 +10,7 @@
 # - Implementing electron and hole ("orbital") degrees of freedom
 #   using different lattices
 
+import _defs
 import kwant
 
 # For plotting
@@ -85,19 +86,24 @@
                     smatrix.transmission(0, 0) +
                     smatrix.transmission(1, 0))
 
-    pyplot.figure()
+    fig = pyplot.figure()
     pyplot.plot(energies, data)
     pyplot.xlabel("energy [t]")
     pyplot.ylabel("conductance [e^2/h]")
-    pyplot.show()
+    pyplot.setp(fig.get_axes()[0].get_xticklabels(),
+               fontsize=_defs.mpl_tick_size)
+    pyplot.setp(fig.get_axes()[0].get_yticklabels(),
+               fontsize=_defs.mpl_tick_size)
+    fig.set_size_inches(_defs.mpl_width_in, _defs.mpl_width_in * 3. / 4.)
+    fig.subplots_adjust(left=0.15, right=0.95, top=0.95, bottom=0.15)
+    for extension in ('pdf', 'png'):
+        fig.savefig("superconductor_transport_result." + extension,
+                    dpi=_defs.dpi)
 
 
 def main():
     syst = make_system()
 
-    # Check that the system looks as intended.
-    kwant.plot(syst)
-
     # Finalize the system.
     syst = syst.finalized()
 
